packages/web-app-files/src/uppyUploadPlugin.ts

```diff
@@ -1,36 +1,74 @@
-import Uppy from '@uppy/core'
+import Uppy, { UppyFile } from '@uppy/core'
 import BasePlugin from '@uppy/core/lib/BasePlugin.js'
-import { dirname } from 'path'
+import filesize from 'filesize'
+import { dirname, join } from 'path'
 import * as uuid from 'uuid'
+import { Language } from 'vue3-gettext'
 import { Ref, unref } from 'vue'
 import { Store } from 'vuex'
 import { RouteLocationNormalizedLoaded } from 'vue-router'
 import { Resource, SpaceResource } from 'web-client/src'
 import { urlJoin } from 'web-client/src/utils'
 import { UppyResource } from 'web-runtime/src/composables/upload'
 import { UppyService } from 'web-runtime/src/services/uppyService'
+import { ResourceConflict } from './helpers/resource'
+import { locationPublicLink } from './router/public'
+import { locationSpacesGeneric } from './router/spaces'
+import { isShareSpaceResource } from 'web-client/src/helpers'
+import { ClientService } from 'web-pkg/types'
+
+export interface HandleUploadOptions {
+  clientService: ClientService
+  hasSpaces: Ref<boolean>
+  language: Language
+  route: Ref<RouteLocationNormalizedLoaded>
+  store: Store<any>
+  uppyService: UppyService
+  id?: string
+  space?: SpaceResource
+  quotaCheckEnabled?: boolean
+  directoryTreeCreateEnabled?: boolean
+  conflictHandlingEnabled?: boolean
+}
 
 /**
- * Prepares all input files and adds necessary data for the upload. This is only used for
- * the cloud import at the moment.
+ * Plugin to handle the file upload with uppy. The process goes through the following steps:
  *
- * It basically does the same as the inputFilesToUppyFiles method from the upload helpers.
- * TODO: Remove the upload helpers and use this plugin for upload handling in general.
+ * 1. convert input files to uppy resources
+ * 2. check quota if spaces are enabled
+ * 3. handle potential naming conflicts
+ * 4. create directory tree if needed
+ * 5. start upload
  */
 export class HandleUpload extends BasePlugin {
```

Comment:
I think it's a bit unfortunate that the file name does not resemble the class name at all.. can we streamline this more?